docs/dev/sql-nested-function-where-clause.md

```diff
@@ -0,0 +1,397 @@
+## Description
+
+The `nested` function when used in the `WHERE` clause of an SQL statement filters documents of `nested` object type based on field properties. Two syntax options are available to users see section [1.1 Syntax](#11-syntax), both syntax options achieve the goal of filtering with a `nested` field, an `Operator` and a `Literal` value. Both syntax options produce different DSL to push to OpenSearch based on the `AND/OR/NOT` boolean queries residing inside a `nested` query or vice versa. See section [4?]() for examples depicting this alternate DSL creation. Testing large groups of push down execution times for identical queries using the different syntax options showed a nominal difference of approximately 3% for Syntax option 2.
```

Comment:
Something just sounds slightly off in the grammar off:
`Two syntax options are available to users see section [1.1 Syntax]`

Maybe add a comma to make it flow better?
`Two syntax options are available to users, see section [1.1 Syntax]`